mono/utils/mono-rand-windows.c

```diff
@@ -163,6 +92,5 @@ mono_rand_try_get_bytes (gpointer *handle, guchar *buffer, gint buffer_size, Mon
 void
 mono_rand_close (gpointer handle)
 {
-	mono_rand_win_close_provider ((MONO_WIN32_CRYPT_PROVIDER_HANDLE) handle);
```

Comment:
Maybe assert that the handle value is the same as returned from mono_rand_init?

Reply:
No need.

Reply:
Maybe just as a precautions if the methods get called with incorrect handle value since we know what it should be.

Reply:
It can also be 0. And eventually it'll be ignored. But ok.

You think anyone depends on seedability? I think it is ignored on newer OS but I'll check the docs again.

Reply:
In terms of the provider value, I think some defense-in-depth has value, and some is pointless.
In this case, pretty pointless. But we can only assert 0 or constant for win7 or newer, which won't be active for now. For Vista support, the value is dynamic.